### PR TITLE
If a user doesnt have the donator role when making a donator server, give it to them

### PR DESCRIPTION
**Pull request by Deveroonie**

### CODE REVIEW

**Comment by danielpmc** [comment]
I will leave this until im able to test it. Should be in a few hours

**Comment by itzJOHv** [comment]
y do u use roles.cache.find() if u can directly do roles.cache.get()

**Comment by SirEpiclyAmazing** [comment]
> I will leave this until im able to test it. Should be in a few hours

3 days later.....................................................

**Comment by SirEpiclyAmazing** [comment]
dan wont accept it so closed

**Comment by itzJOHv** [comment]
why the fuck did u say that this is closed then reopened it

**Comment by SirEpiclyAmazing** [comment]
cos i can <3

**Comment by danielpmc** [comment]
Deleted @SirEpiclyAmazing's reply. Leaked unsub token from replying to email

**Comment by SirEpiclyAmazing** [comment]
when are u gonna test it

**Comment by danielpmc** [comment]
Just fixing Node 8 right now, then ill test it after that.

**Comment by SirEpiclyAmazing** [comment]
pog

**Comment by DIBSTERYT** [comment]
_10 Days ago..._

**Comment by danielpmc** [comment]
> Just fixing Node 8 right now, then ill test it after that.

This was checked just after that and there are some issues that will need fixing. Ill submit them here then merge it

**Comment by Deveroonie** [comment]
what are the issues

File: bot/discord/commands/server/create-donator.js [hunk 1]
```diff
@@ -92,7 +92,12 @@ exports.run = async(client, message, args) => {
         serverCreateSettings_Prem.createServer(types[args[1].toLowerCase()])
             .then(response => {
 
-                userPrem.set(message.author.id + '.used', userP.used + 1);
+                userPrem.set(message.author.id + '.used', userP.used + 1).then(() => {
+                 if (!message.member.roles.has(allowedRole.id)) {
```
Comment (itzJOHv) [inline bot/discord/commands/server/create-donator.js:96]:
```suggestion
                 if (!message.member.roles.cache.has(allowedRole.id)) {
```

File: bot/discord/commands/server/create-donator.js [hunk 2]
```diff
@@ -92,7 +92,12 @@ exports.run = async(client, message, args) => {
...
+                     let role = message.guild.roles.cache.find(r => r.id === "898041754564046869");
```
Comment (itzJOHv) [inline bot/discord/commands/server/create-donator.js:97]:
```suggestion
                     const role = message.guild.roles.cache.get("898041754564046869");
```

File: bot/discord/commands/server/create-donator.js [hunk 3]
```diff
@@ -92,7 +92,12 @@ exports.run = async(client, message, args) => {
...
+                     message.author.roles.add(role)
```
Comment (itzJOHv) [inline bot/discord/commands/server/create-donator.js:98]:
```suggestion
                     if (role) message.member.roles.add(role);
```

File: bot/discord/commands/server/create-donator.js [hunk 4]
```diff
@@ -103,7 +108,7 @@ exports.run = async(client, message, args) => {
                     //.addField(`__**Node:**__`, "Node 7 - Boosters/Donators")
                     .addField(`__**WARNING**__`, `**DO NOT USE JAVA TO RUN GAMESERVERS. IF THERE IS A GAME YOU ARE WANTING TO HOST AND IT DOES NOT HAVE A SERVER PLEASE MAKE A TICKET**`)
                 message.channel.send(embed)
-
+
```
Comment (itzJOHv) [inline bot/discord/commands/server/create-donator.js:111]:
```suggestion

```

**Review by itzJOHv — CHANGES_REQUESTED** [review]

**Comment by WilliamDavidHarrison** [comment]
Some people don’t donate though, they get given the servers from giveaways, etc.

**Comment by Sliden101** [comment]
mfw http://cdn.sliden.tech/MLh99A

**Comment by Deveroonie** [comment]
I'm not fixing this as I was banned from the DBH Discord.

**Comment by itzJOHv** [comment]
> I'm not fixing this as I was banned from the DBH Discord.

L

**Comment by HyptexPvP** [comment]
> I'm not fixing this as I was banned from the DBH Discord.

That's crazy

**Comment by danielpmc** [comment]
Useless PR. Thread locked

**Pull request closed without merging**